### ISSUE TITLE
Style/Lambda with default settings produce wrong output

### ISSUE THREAD

**Issue by kvokka** [rubocop/rubocop#9555]
## Expected behavior

it should replace `lambda { ... }` with `lambda do ... end` regarding the cop [description](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/Lambda) 

## Actual behavior

no changes

## Steps to reproduce the problem

cop `Style/Lambda` with default settings:

```yaml
Style/Lambda:
  EnforcedStyle: line_count_dependent
```

code sample
```ruby
config.action_dispatch.hosts_response_app = lambda  { |_env|
'42'
'55'
}
```

run `bundle exec rubocop -a`

## RuboCop version

```
1.11.0 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 2.7.2 x86_64-darwin19)
```

**Comment by dvandersluis** [comment]
`Style/Lambda` does not change braces, it just enforces proper use of `->` vs the `lambda` keyword.

`Style/BlockDelimiters` _can_ be used for lambdas, but they are ignored by default:
https://github.com/rubocop/rubocop/blob/04019d2aa705ad2ca3e9ba43053284a5277fdd63/config/default.yml#L2836-L2852

You can override the `IgnoredMethods` configuration for that cop if you're so included (but mind the note).
